src/scripts/mod.rs

```diff
@@ -0,0 +1,309 @@
+                ExitStatus::Exited(c) => c as i32,
+                ExitStatus::Signaled(c) => c as i32,
+                ExitStatus::Other(c) => c,
+                _ => -1,
+            };
+            if es != 0 {
+                return Err(anyhow!("Exit code = {}", es));
+            }
+            Ok(c.stdout_str())
+        }
+        Err(error) => {
+            debug!("Command error {}", error.to_string());
+            return Err(anyhow!(error.to_string()));
+        }
+    }
+}
+
+pub fn find_scripts(mut path: PathBuf) -> Result<Vec<PathBuf>> {
+    path.push(".rustscan_scripts");
+    // check if dir is available
+    if path.is_dir() {
+        debug!("Scripts folder found {}", &path.display());
+        let mut files_vec: Vec<PathBuf> = Vec::new();
+        for entry in fs::read_dir(path)? {
+            let entry = entry?;
+            files_vec.push(entry.path());
+        }
+        return Ok(files_vec);
+    } else {
+        return Err(anyhow!("cant find scripts folder"));
+        // println!("Do you want me to create the {} directory?", &scripts_dir.to_string_lossy());
+        // std::fs::create_dir(&scripts_dir).unwrap();
+    }
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct ScriptFile {
+    pub path: Option<PathBuf>,
+    pub tags: Option<Vec<String>>,
+    pub developer: Option<Vec<String>>,
+    pub execute_with: Option<String>,
+    pub execute_with_path: Option<PathBuf>,
+    pub join_ports: Option<String>,
+    pub call_format: Option<String>,
+    pub port: Option<String>,
+}
+
+pub fn parse_scripts(scripts: Vec<PathBuf>) -> Vec<ScriptFile> {
+    let mut parsed_scripts: Vec<ScriptFile> = Vec::with_capacity(scripts.len());
+    for script in scripts {
+        let real_path = script.clone();
+        let mut lines_buf = String::new();
+        if let Ok(file) = File::open(script) {
+            for line in io::BufReader::new(file).lines().skip(1) {
+                if let Ok(mut line) = line {
+                    if line.starts_with("#") && line.len() > 3 {
+                        line.retain(|c| c != '#');
+                        line = line.trim().to_string();
+                        line.push_str("\n");
+                        lines_buf.push_str(&line);
+                    } else {
+                        break;
+                    }
+                }
+            }
+        } else {
+            debug!("Failed to read file: {}", &real_path.display());
+            continue;
+        }
+        debug!("ScriptFile lines\n{}", &lines_buf);
+        if let Ok(mut parsed) = toml::from_str::<ScriptFile>(&lines_buf) {
+            debug!("Parsed ScriptFile {:?}", &parsed);
+            parsed.path = Some(real_path);
+            parsed_scripts.push(parsed);
+        } else {
+            debug!("Failed to parse toml");
+        }
+    }
+    parsed_scripts
+}
+
+/*
+Example ScriptFile headers, all fields are optional but we have to do the checks if something is necessary, for example an executor path.
```

Comment:
Do we need to move this to `ScriptFile` instead of `ScriptConfig` and document `ScriptConfig` itself?